editor/editor_file_system.cpp

```diff
@@ -1894,31 +2088,34 @@ HashSet<String> EditorFileSystem::get_valid_extensions() const {
 	return valid_extensions;
 }
 
-void EditorFileSystem::register_global_class_script(const String &p_search_path, const String &p_target_path) {
+void EditorFileSystem::_register_global_class_script(const String &p_search_path, const String &p_target_path, const String &p_type, const String &p_script_class_name, const String &p_script_class_extends, const String &p_script_class_icon_path) {
 	ScriptServer::remove_global_class_by_path(p_search_path); // First remove, just in case it changed
 
-	int index = -1;
-	EditorFileSystemDirectory *efd = find_file(p_search_path, &index);
-
-	if (!efd || index < 0) {
-		// The file was removed
-		return;
-	}
-
-	if (!efd->files[index]->script_class_name.is_empty()) {
+	if (!p_script_class_name.is_empty()) {
 		String lang;
 		for (int j = 0; j < ScriptServer::get_language_count(); j++) {
-			if (ScriptServer::get_language(j)->handles_global_class_type(efd->files[index]->type)) {
+			if (ScriptServer::get_language(j)->handles_global_class_type(p_type)) {
 				lang = ScriptServer::get_language(j)->get_name();
+				break;
 			}
 		}
 		if (lang.is_empty()) {
 			return; // No lang found that can handle this global class
 		}
 
-		ScriptServer::add_global_class(efd->files[index]->script_class_name, efd->files[index]->script_class_extends, lang, p_target_path);
-		EditorNode::get_editor_data().script_class_set_icon_path(efd->files[index]->script_class_name, efd->files[index]->script_class_icon_path);
-		EditorNode::get_editor_data().script_class_set_name(p_target_path, efd->files[index]->script_class_name);
+		ScriptServer::add_global_class(p_script_class_name, p_script_class_extends, lang, p_target_path);
+		EditorNode::get_editor_data().script_class_set_icon_path(p_script_class_name, p_script_class_icon_path);
+		EditorNode::get_editor_data().script_class_set_name(p_target_path, p_script_class_name);
+	}
+}
+
+void EditorFileSystem::register_global_class_script(const String &p_search_path, const String &p_target_path) {
+	int index_file;
+	EditorFileSystemDirectory *efsd = find_file(p_search_path, &index_file);
+	if (efsd) {
+		EditorFileSystem::get_singleton()->_register_global_class_script(p_search_path, p_target_path, efsd->files[index_file]->type, efsd->files[index_file]->script_class_name, efsd->files[index_file]->script_class_extends, efsd->files[index_file]->script_class_icon_path);
+	} else {
+		EditorFileSystem::get_singleton()->_register_global_class_script(p_search_path, p_target_path, "", "", "", "");
```

Comment:
This will just remove the script, no? You could call `remove_global_class_by_path()` directly.

Reply:
Done!